google-cloud-spanner/src/main/java/com/google/cloud/spanner/DatabaseInfo.java

```diff
@@ -141,6 +155,29 @@ public Builder setDialect(Dialect dialect) {
       return this;
     }
 
+    @Override
+    public Builder setProtoDescriptors(@Nonnull byte[] protoDescriptors) {
+      Preconditions.checkNotNull(protoDescriptors);
+      this.protoDescriptors = ByteString.copyFrom(protoDescriptors);
+      return this;
+    }
+
+    @Override
+    public Builder setProtoDescriptors(@Nonnull InputStream inputStream) throws IOException {
+      Preconditions.checkNotNull(inputStream);
+      byte[] byteArray = ByteStreams.toByteArray(inputStream);
+      this.protoDescriptors = ByteString.copyFrom(byteArray);
+      return this;
+    }
+
+    @Override
+    public Builder setProtoDescriptors(String filePath) throws IOException {
+      Preconditions.checkState(filePath.length() != 0, "Input File Path cannot be empty.");
+      InputStream inputStream = getClass().getClassLoader().getResourceAsStream(filePath);
```

Comment:
It looks like the method only support reading files from `Project Resources`. If yes please clarify it in java doc.

Reply:
Added in java doc

Reply:
Is Project Resources a good default?

Reply:
Yes. Having the resources with in the class path can help users provide a relative file path as input and Project resources directory is a good place to maintain all these files.

Reply:
@olavloite 
My understanding here is class loader can find a resource if it is there in the classpath and resources directory would always be a part of the classpath. I tried this with a small demo application and it works fine.
However, I am not aware of the behaviour if there are any custom/multiple class loaders. I would need some time to analyse the behaviour in this case.
In the interest of time, I am planning to remove this overload that accepts a file path as argument and take it separately in another PR.
Can you please let me know your views here?

Reply:
I think that removing this for now is a good choice. People can easily use resources from their class path using the overload that takes an `InputStream` by calling `getResourceAsStream()` themselves.